Example/VialerSIPLib/AppDelegate.swift

```diff
@@ -125,30 +120,20 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     }
 
     func displayIncomingCall(call: VSLCall) {
-        if #available(iOS 10, *) {
-            DDLogWrapper.logInfo("Incoming call block invoked, routing through CallKit.")
-            providerDelegate?.reportIncomingCall(call)
-        } else {
-            DDLogWrapper.logInfo("Incoming call block invoked, using own app presentation.")
-            NotificationCenter.default.post(name: AppDelegate.Configuration.Notifications.incomingCall,
-                                            object: self,
-                                            userInfo: [VSLNotificationUserInfoCallKey : call])
-        }
+        DDLogWrapper.logInfo("Incoming call block invoked, routing through CallKit.")
+        providerDelegate?.reportIncomingCall(call)
     }
 
     // MARK: - CallKit outbound call from a iOS native view e.g. Contacts or Recents.
     func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([Any]?) -> Void) -> Bool {
-        if #available(iOS 10.0, *) {
-            guard let handle = userActivity.startCallHandle else {
-                return false
-            }
-
-            VialerSIPLib.sharedInstance().callManager.startCall(toNumber: handle, for:account, completion: { (call, error) in
-                if error != nil {
-                    DDLogWrapper.logError("Could not create outbound call. Error: \(error!)")
-                }
-            })
+        guard let handle = userActivity.startCallHandle else {
+            return false
         }
+        VialerSIPLib.sharedInstance().callManager.startCall(toNumber: handle, for:account, completion: { (call, error) in
+            if error != nil {
+                DDLogWrapper.logError("Could not create outbound call. Error: \(error!)")
```

Comment:
should be return false on error?

Reply:
This is the example app of the VialerSIPLib. The code did not return false on the error before the last change. It seems logical what you proposing but I do not want to include this change now or invest more time for this. The number one priority is the release of vialer with the new VialerSIPLib. My opinion for code improvements like this one is to keep them in mind or in a ticket and fully develop them in a separate branch later when the priorities are suitable for it. Otherwise we will never release and we will continue to make infinite small code improvements until the end of time.